iscript/src/iscript/mac.py

```diff
@@ -279,11 +293,13 @@ async def sign_app(sign_config, app_path, entitlements_path, provisioning_profil
 
         for file_ in files:
             abs_file = os.path.join(top_dir, file_)
+            sign_command = _get_sign_command(identity, keychain, sign_config, file_, entitlements_path)
             await _do_sign_file(top_dir, abs_file, file_, sign_command, app_path_len, app_executable)
 
-    await sign_libclearkey(contents_dir, sign_command, app_path)
+    await sign_libclearkey(contents_dir, _get_sign_command(identity, keychain, sign_config), app_path)
 
     # sign bundle
+    sign_command = _get_sign_command(identity, keychain, sign_config, entitlements_path=entitlements_path)
```

Comment:
Will we ever need to not use an entitlements file or hardened runtime on the top level app?

Reply:
I don't know! I suspect not. I didn't think a YAGNI philosophy would make sense here though, since it was trivial to support all paths.